src/parserfunc.c

```diff
@@ -866,7 +866,7 @@ pf_do_not(ParseNode* self)
     {
         set_error(self->state, PARSER_ERR_OVERFLOW, NULL);
         free(ans);
-        ans = NULL;
+        return NULL;
     }
     mp_not(val, self->state->options->wordlen, ans);
```

Comment:
```suggestion
    if (ans)
        mp_not(val, self->state->options->wordlen, ans);
```

Reply:
or just `free(val)` before the return. this is what all the other cases to